File: 0217_contains_duplicate/ContainDuplicate.py
class ContainsDuplicate(object):
    def containsDuplicate(self, nums):
        map = {}
        
        for i, num in enumerate(nums):
            if num in map:
                return True
            map[num] = i
        return False

ob = ContainsDuplicate()

nums = [1, 2, 3, 1]

answer = ob.containsDuplicate(nums)

print(f"Answer {answer}")

File: 0217_contains_duplicate/ContainDuplicate.php
<?php 

class ContainDuplicate {

    /**
     * @param Integer[] $nums
     * @return Boolean
     */
    function containsDuplicate($nums) {
        $map = [];
        
        foreach($nums AS $index => $num){
            if(isset($map[$num])){
                return true;
            }

            $map[$num] = $index;
        }

        return false;
    }
}

$nums = [1, 2, 3, 1];
$ob = new ContainDuplicate();
echo $ob->containsDuplicate($nums) ? "True" : "False";
echo "\n";